compiler-core/src/build/package_loader.rs

```diff
@@ -1715,3 +1682,143 @@ impl<'a> Inputs<'a> {
         Ok(())
     }
 }
+
+/// A Gleam source file (`.gleam`) and the module name deduced from it
```

Comment:
For everything downwards from here, I'm not sure if this is the best module, or the best location in the source file to put these.